lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
Hmm, darn.  For what it's worth, here's a bit of discussion about the commit that caused this functionality change:

https://reviews.llvm.org/D26594

Maybe the information there will help understand the changes and how to accommodate them properly.

I've run into this while migrating another project to 4.0 but haven't had time to sort out the changes sorry :(.